File: src/test/testdata/inspections/flushInsideLoop/Test5.java
package flushInsideLoop;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.MathUtils;

class ClassA extends SpriteBatch {

}

class ClassB extends ClassA {
    @Override
    public void flush() {
        super.flush();
    }
}

class ClassC extends ClassB {

    @Override
    public void flush() {

    }

    public void myFlush() {
        super.flush();
    }

    static void test() {

        ClassA a = new ClassA();
        ClassB b = new ClassB();
        ClassC c = new ClassC();

        while (MathUtils.randomBoolean()) {
            <warning>a.flush()</warning>;
            <warning>b.setTransformMatrix(null)</warning>;
            <warning>b.flush()</warning>;
            c.flush();
            <warning>c.myFlush()</warning>;
        }


    }
}

File: src/test/testdata/inspections/flushInsideLoop/Test5.kt
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.MathUtils

open class A: SpriteBatch() {

  override fun flush() {
    super.flush()
  }
}

open class B: A() {
  override fun flush() {
    super.flush()
  }
}

class C: B() {
  override fun flush() {

  }

  fun myFlush() {
    super.flush()
  }
}

fun main() {
  while(MathUtils.randomBoolean()) {
    A().<warning>flush()</warning>
    B().<warning>flush()</warning>
    C().flush()
    C().<warning>myFlush()</warning>

    B().let {
      it.<warning>flush()</warning>
    }

    B().apply { <warning>flush()</warning> }

    val a = A()
    val b = a
    val c = b

    c.let {
      it.apply {
        <warning>flush()</warning>
      }
    }
  }
}